controllers/pvc-mutator/encryption/keygen_test.go

```diff
@@ -0,0 +1,388 @@
+package encryption
+
+import (
+	"context"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/storageos/api-manager/controllers/pvc-mutator/encryption/keys"
+	"github.com/storageos/api-manager/internal/pkg/provisioner"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestMutatePVC(t *testing.T) {
+	t.Parallel()
+
+	// Test values only.
+	storageosProvisioner := provisioner.DriverName
+
+	// Create a new scheme and add all the types from different clientsets.
+	scheme := runtime.NewScheme()
+	if err := kscheme.AddToScheme(scheme); err != nil {
+		t.Fatal(err)
+	}
+
+	// StorageOS StorageClass.
+	stosSC := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "stos",
+		},
+		Provisioner: storageosProvisioner,
+	}
+
+	// Non-StorageOS StorageClass.
+	notstosSC := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "non-stos",
+		},
+		Provisioner: "foo-provisioner",
+	}
+
+	testNamespace := "default"
+
+	testcases := []struct {
+		name                              string
+		namespace                         string
+		notStos                           bool
+		betaAnnotation                    bool
```

Comment:
Looks like none of the test cases use `notStos` and `betaAnnotation`. It'd be nice to test them as well, to ensure that the mutation doesn't fail for them due to some reason.

Reply:
`notStos` is useful and used one of the PR-s opened on top of this!